### PR TITLE
Bump to nightly F* version

### PR DESCRIPTION
**Pull request by gebner**
For our upcoming F* dataset, we want to build all F* projects with the same, nightly F* version.  This PR makes the changes necessary to build with the latest and greatest F*.

Some of the tests still fail however, e.g. Z3 fails to show that the invariant is preserved in `invariantDecl_MaintainedIfStmtSatisfies`.

### CODE REVIEW

File: experimental/lib/MyProgramProof.fst [hunk 1]
```diff
@@ -9,7 +9,7 @@ open Armada.Statement
 open Armada.Transition
 open Armada.Type
 open FStar.List.Tot.Base
-open FStar.Tactics
+open FStar.Tactics.V2
```
Comment (gebner) [inline experimental/lib/MyProgramProof.fst:12]: @mtzguido From what I can tell, you've been migrating a lot of code in F* to V2.  Is this what we should do here as well?

Comment (mtzguido) [reply experimental/lib/MyProgramProof.fst:12]: Hi Gabriel. Yes, these changes look perfect to me

**Comment by gebner** [comment]
> Some of the tests still fail however, e.g. Z3 fails to show that the invariant is preserved in `invariantDecl_MaintainedIfStmtSatisfies`.

FWIW, this test (as well as a few other ones) already fail on the `main` branch.

**Comment by LukeXuan** [comment]
> > Some of the tests still fail however, e.g. Z3 fails to show that the invariant is preserved in `invariantDecl_MaintainedIfStmtSatisfies`.
> 
> FWIW, this test (as well as a few other ones) already fail on the `main` branch.

As I understand, the fst files (mostly generated) in the `Tests` folder are still unstable and it's fine they are failing. The files in the `lib` folder should be verified without any problem.

**Comment by LukeXuan** [comment]
Also, I'm not sure if we can backport the changes to our development repo, as we would prefer a stable fstar version for benchmarking, etc. But the changes doesn't seem to complicated and maybe we can just merge every time we are pushing to the main branch here.

**Comment by gebner** [comment]
> we would prefer a stable fstar version for benchmarking, etc.

Would it be okay to track the latest F* release instead, or are you looking for longer-term stability?  They come out every couple of months (last one was actually on Saturday).

**Comment by jaylorch** [comment]
Yes, tracking the latest F* release would be preferable to tracking the nightly F* version. Is that OK?

**Comment by gebner** [comment]
Yes, that's great.  We can make a branch if we need further changes to support nightly F*, but that's going to be a much smaller delta.  Starmada was three releases worth of breaking changes behind.

**Comment by jaylorch** [comment]
OK, sounds good. Could you please change your PR to refer to v2024.01.13 instead of that specific nightly commit, then? (e.g., in docker/Dockerfile).

**Pull request merged**